modules/listener/listener.tf

```diff
@@ -0,0 +1,191 @@
+data "aws_iam_policy_document" "listener" {
+  statement {
+    actions = [
+      "logs:DescribeSubscriptionFilters",
+      "logs:DeleteSubscriptionFilter",
+      "logs:PutSubscriptionFilter",
+      "logs:DescribeLogGroups",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "lambda:AddPermission",
+      "lambda:RemovePermission",
+      "lambda:InvokeFunction",
+      "lambda:GetFunction",
+      "logs:DescribeLogStreams",
+      "logs:DescribeSubscriptionFilters",
+      "logs:FilterLogEvents",
+      "logs:GetLogEvents",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = ["*"]
+  }
+}
+
+resource "aws_lambda_function" "listener" {
+  s3_bucket     = var.mode == "dev" ? "axiom-cloudformation-dev" : "axiom-cloudformation"
+  s3_key        = "axiom-cloudwatch-forwarder/v1.2.0/forwarder.zip"
+  function_name = format("%s-listener", var.prefix)
+  description   = "Axiom CloudWatch Automatic log groups listener lambda"
+  logging_config {
+    log_format = "JSON"
+    log_group  = aws_cloudwatch_log_group.listener.name
+  }
+  handler = "listener.lambda_handler"
+  runtime = "python3.9"
+  role    = aws_iam_role.listener.arn
+  timeout = 300
+
+  environment {
+    variables = {
+      "AXIOM_CLOUDWATCH_FORWARDER_LAMBDA_ARN" = var.forwarder_lambda_arn
+      "LOG_GROUP_PREFIX" : var.log_groups_prefix
+    }
+  }
+
+  tags = {
+    PartOf    = var.prefix
+    Platform  = "Axiom"
+    Component = "axiom-cloudwatch-listener"
+  }
+}
+
+resource "aws_lambda_permission" "allow_cloudwatch" {
+  count         = var.enable_cloudtrail ? 1 : 0
```

Comment:
I think this does not depend on `enable_cloudtrail`. The listener always triggers on some events by Event Bridge right?

Reply:
This part confuses me, Event Bridge depends on Cloudtrail to trigger those events, right? When I tested without Cloudtrail the triggering didn't happen.

Reply:
Yes, this permission must be created, but it should not be dependant on this flag, as without this permission it will not work.